src/server/cluster/cluster_family.cc

```diff
@@ -413,19 +434,40 @@ void ClusterFamily::DflyClusterConfig(CmdArgList args, ConnectionContext* cntx)
     return rb->SendError("Invalid JSON cluster config", kSyntaxErrType);
   }
 
-  auto deleted_slot_ids = cluster_config_->SetConfig(json.value());
-  if (!deleted_slot_ids.has_value()) {
+  unique_ptr<ClusterConfig> new_config =
+      ClusterConfig::CreateFromConfig(server_family_->master_id(), json.value());
+  if (new_config == nullptr) {
+    LOG(WARNING) << "Can't set cluster config";
     return rb->SendError("Invalid cluster configuration.");
   }
 
+  lock_guard gu(set_config_mu);
+
+  bool is_first_config = true;
+  SlotSet before;
+  if (tl_cluster_config != nullptr) {
+    is_first_config = false;
+    before = tl_cluster_config->GetOwnedSlots();
+  }
+
+  auto cb = [&](util::ProactorBase* pb) {
+    tl_cluster_config = make_unique<ClusterConfig>(*new_config);
```

Comment:
When moving to thread local config, because this is not atomic, we might see one thread showing config different than other thread. But I dont think this is a problem. What do you think?

Reply:
I think that to some degree this problem exists even with a global config (like switching between yielding), but I think it's ok..
The least of our concerns when talking about migration side effects :)